src/metatensor/models/utils/data/writers/xyz.py

```diff
@@ -75,50 +66,48 @@ def write_xyz(
 
             for gradient_name, gradient_block in block.gradients():
                 # here, we assume that gradients are always an array, and never a scalar
-                if capabilities.outputs[target_name].quantity == "energy":
-                    if gradient_name == "positions":
-                        if only_one_energy:
-                            name_for_saving = "forces"
-                        else:
-                            name_for_saving = f"forces[{target_name}]"
-                        arrays[name_for_saving] = (
-                            # squeeze the property dimension
-                            -gradient_block.values.detach()
-                            .cpu()
-                            .squeeze(-1)
-                            .numpy()
-                        )
-                    elif gradient_name == "strain":
-                        strain_derivatives = (
-                            # squeeze the property dimension
-                            gradient_block.values.detach()
-                            .cpu()
-                            .squeeze(-1)
-                            .numpy()
+                internal_name = f"{target_name}_{gradient_name}_gradients"
+                external_name = to_external_name(internal_name, capabilities.outputs)
+
+                if "forces" in external_name:
+                    arrays[external_name] = (
+                        # squeeze the property dimension
+                        -gradient_block.values.detach()
+                        .cpu()
+                        .squeeze(-1)
+                        .numpy()
+                    )
+                elif "virial" in external_name:
```

Comment:
where do I see that we write the stress and the virial if there is a gradient present in the traget TensorMap.

Reply:
https://github.com/lab-cosmo/metatensor-models/blob/3f012d18bbe287463886ead47d71a51a11317d78/src/metatensor/models/utils/data/writers/xyz.py#L101
Both stresses and virials are registered in the "info" here
And then the info will be printed by ASE as part of the atoms object in the xyz file